test/ex_doc/formatter/epub_io_test.exs

```diff
@@ -0,0 +1,55 @@
+defmodule ExDoc.Formatter.EPUBIOTest do
```

Comment:
I think we can have a much smaller and simpler test.

If we avoid fixtures then we don't need to worry about warnings and any of that either. In fact, it's totally fine if we point ExDoc to a directory with no .beam files (the directory doesn't even have to exist.) 

Instead of refuting that a given warning was emitted, let's assert there were no warnings.

How about this?

```elixir
defmodule ExDoc.Formatter.EPUBIOTest do
  use ExUnit.Case, async: false

  @moduletag :tmp_dir

  test "does not warn if writing to an empty existing directory", %{tmp_dir: tmp_dir} do
    File.mkdir!("#{tmp_dir}/doc")

    assert ExUnit.CaptureIO.capture_io(:stderr, fn ->
             generate_docs(tmp_dir)
           end) == ""
  end

  test "warns if trying to write into existing directory with files", %{tmp_dir: tmp_dir} do
    File.mkdir!("#{tmp_dir}/doc")
    File.touch!("#{tmp_dir}/doc/foo.txt")

    assert ExUnit.CaptureIO.capture_io(:stderr, fn ->
             generate_docs(tmp_dir)
           end) =~ "ExDoc is outputting to an existing directory"

    # don't warn twice
    assert ExUnit.CaptureIO.capture_io(:stderr, fn ->
             generate_docs(tmp_dir)
           end) == ""
  end

  defp generate_docs(tmp_dir) do
    config = [
      app: :foo,
      formatter: "epub",
      output: "#{tmp_dir}/doc",
      source_beam: "#{tmp_dir}/ebin"
    ]

    ExDoc.generate_docs("Foo", "1.0.0", config)
  end
end
```

Reply:
Thanks for the suggestions! I've applied them and those tests are indeed much cleaner now.